rfcs/datepicker-component.md

```diff
@@ -0,0 +1,148 @@
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => <DatePicker />;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => {
+  return <DatePicker
+    initialState={{
+      value: [new Date(2018, 1, 15), new Date(2018, 8, 25)]
+    }}
+    isRange={true}
+    dateFormat={'MM/DD/YY'}
+    minDate={new Date(2000, 1, 1)}
+    maxDate={new Date(2020, 1, 1)}
+    monthRange={2}
+    onChange={(event, value)=>{console.log(value)}}
+    overrides={{
+      MonthGridDay: props => <div style={{
+        color: props.$inRange ? 'red' : 'blue',
+        borderRadius: '5px',
+      }}>{props.$value}</div>,
+    }}
+    />
+}
+```
+
+## Exports
+
+* `DatePicker`
+* `StatefulDatePicker`
+* `StatefulDatePickerContainer`
+* `StyledRoot`
+* `StyledDateLabel`
+* `StyledDateInput`
+* `StyledMonthAndYearBar`
+* `StyledMonthSelect`
+* `StyledYearSelect`
+* `StyledMonthGrid`
+* `StyledMonthGridLabel`
+* `StyledMonthGridDay`
+* `StyledQuickSelect`
+
+## `DatePicker` API
+
+* `isRange: boolean` - Optional
+  If set, datepicker is to select 2 dates range, so there are two date input and `value` comes as Array of two dates, min and max. Otherwise only one date is selected. Default is `false`
+* `dateFormat: string` - Optional
+  Date format that is accepted by `Date.parse` function. RFC2822 tools.ietf.org/html/rfc2822#section-3.3 standard. Default is `DD/MM/YYYY`
+* `value: Date | Array<Date>` - Optional
+  Selected `Date`. If `isRange` is set, `value` is an array of 2 values.
+* `minDate: Date` - Optional
+  minimum date available for selection. Default is `Jan 01 1970`
+* `maxDate: Date` - Optional
+  maximum date available for selection. Default is current date + 3 years.
+* `disabled: boolean`:
+  Disable control from being changed
+* `monthRange: number`
+  Amount of minth grids to be shown. Accepts 1 or 2. Default is `1`
+* `overrides: {Root: (props: {[string]: any}) => React$Node, DateLabel: (props: {[string]: any}) => React$Node,
+ DateInput: (props: {[string]: any}) => React$Node, MonthAndYearBar: (props: {[string]: any}) => React$Node,
+  MonthSelect: (props: {[string]: any}) => React$Node, YearSelect: (props: {[string]: any}) => React$Nod,
+  MonthGrid: (props: {[string]: any}) => React$Node, MonthGridLabel: (props: {[string]: any}) => React$Node,
+  MonthGridDay: (props: {[string]: any}) => React$Node, QuickSelect: (props: {[string]: any}) => React$Node}`
+  * `Root` - parent element of whole datepicker
+  * `DateLabel` - label above date input
+  * `DateInput` -  date input showing current selected date
+  * `MonthAndYearBar` - a bar\panel showing current selected month and year
+  * `MonthSelect` - select component to pick a month
+  * `YearSelect` - select component to pick a year
+  * `MonthGrid` - a grid of days in month
+  * `MonthGridLabel` - label for day of week in grid
+  * `MonthGridDay` - day number in a grid
```

Comment:
yes, it's a default behavior. So range consists of one date only and inRange should be `true` for it. Do you have access to UX design of BaseUI?
cc @nguyensomniac

Reply:
I see, is there way if we want to highlight one date without making that minDate-maxDate? We probably can have some use case to customize highlighted date cell by date, if we can have access to date on MonthGridDay, what's your thought on here?

Reply:
min and max dates are not for the range mode, they only indicate how far you can swipe left and right to pick the date. `isRange` actually indicates what would be the pick up, is it one date or a range, so highlight works correspondingly. Currently we expose next thisn to `MonthGridDay`. It might be enough to calculate exact date as you may also need to know month and year. However, if you see issues with props below, we can add more

* `$inRange: boolean`
  For `MonthGridDay` only. True if day is in range between min and max selected date value
* `$value: number`
  For `MonthGridDay` only. Number of day in month

Reply:
Yes, I wonder if we can support highlight a exact date without selecting? I see we can definitely highlight the date or range we've selected, but want to see if we can highlight without selecting.

On our service, we often have use case for 'recommending' a date by highlighting a date - which is not necessarily today or any calculable date but it can be a random date.

Reply:
would you like to highlight with another color?
so those date will have $value for day of month and $inRange will be false
It still should be enough for override logic to make it happen.
@nguyensomniac  do you want to add it as default feature?

Reply:
I think it makes sense to highlight a specific day, a good example of the day highlighting would be a current day and maybe we can make it customizable. The highlighted should probably have a separate style from what is selected since `hightlighted` doesn't mean `selected`.

Reply:
Yes it would be different color other than color for 'selected'. Correct, highlighing today is one example, but we can also have different date highlighted.